pkg/operator/controller/status/controller.go

```diff
@@ -421,23 +421,32 @@ func checkGatewayAPICRDs(gatewayAPICRDs []apiextensionsv1.CustomResourceDefiniti
 		return true, "No Gateway API CRDs found, upgradeable by default."
 	}
 
-	managedCRDNames := map[string]bool{}
+	managedCRDMap := make(map[string]*apiextensionsv1.CustomResourceDefinition)
 	for _, crd := range managedGatewayAPICRDs {
-		managedCRDNames[crd.Name] = true
+		managedCRDMap[crd.Name] = crd
 	}
 
 	nonStandardCRDs := []string{}
+	incompatibleCRDs := []string{}
+
 	for _, crd := range gatewayAPICRDs {
-		if !managedCRDNames[crd.Name] {
+		expectedCRD, exists := managedCRDMap[crd.Name]
+		if !exists {
 			nonStandardCRDs = append(nonStandardCRDs, crd.Name)
+		} else if !cmp.Equal(crd, *expectedCRD, cmpopts.EquateEmpty()) {
```

Comment:
hm... I see this file is already using `cmp`, but [its docs](https://pkg.go.dev/github.com/google/go-cmp/cmp#pkg-overview) say "It is intended to only be used in tests, as performance is not a goal and it may panic if it cannot compare the values. Its propensity towards panicking means that its unsuitable for production environments where a spurious panic may be fatal." (Upstream k8s is in the process of getting rid of all non-test uses of `cmp`.)

The standard k8s way to do this would be `apiequality.Semantic.DeepEqual` from `"k8s.io/apimachinery/pkg/api/equality"`.